Mathlib/Analysis/SpecialFunctions/Log/Summable.lean

```diff
@@ -0,0 +1,66 @@
+/-
+Copyright (c) 2024 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck
+-/
+
+import Mathlib.Analysis.SpecialFunctions.Complex.LogBounds
+import Mathlib.Analysis.NormedSpace.FunctionSeries
+
+/-!
+# Summable logs
+
+We give conditions under which the logarithms of a summble sequence is summable. We also give some
+results about when the sums converge uniformly.
+
+TODO: Generalise the indexing set from ℕ to some arbitrary type, but this needs
+Summable.tendsto_atTop_zero to first be done.
+
+-/
+
+open Filter Function Complex Real
+
+open scoped Interval Topology BigOperators Nat Classical Complex
+
+lemma Complex.log_of_summable {f : ℕ → ℂ} (hf : Summable f) :
+    Summable (fun n : ℕ => Complex.log (1 + f n)) := by
+  have hff := Summable.const_smul ((3 : ℝ) / 2) (summable_norm_iff.mpr hf)
+  have := Metric.tendsto_atTop.mp (Summable.tendsto_atTop_zero ((summable_norm_iff.mpr hf)))
+  apply Summable.of_norm_bounded_eventually_nat (fun n => 3/2 * Complex.abs (f n)) hff
+  simp only [smul_eq_mul, gt_iff_lt, ge_iff_le, dist_zero_right, Real.norm_eq_abs, Complex.abs_abs,
+    Complex.norm_eq_abs, eventually_atTop] at *
+  obtain ⟨n, hn⟩ := this (1/2) (one_half_pos)
+  exact Exists.intro n fun m hm ↦ norm_log_one_add_half_le_self (LT.lt.le (hn m hm))
+
+lemma Real.log_of_summable {f : ℕ → ℝ} (hf : Summable f) :
+     Summable (fun n : ℕ => log (1 + |f n|)) := by
+  have : Summable (fun n ↦ Complex.ofRealCLM (log (1 + |f n|))) := by
+    convert Complex.log_of_summable (Complex.ofRealCLM.summable hf.norm) with x
+    rw [ofRealCLM_apply, ofReal_log (by positivity)]
+    simp only [ofReal_add, ofReal_one, norm_eq_abs, ofRealCLM_apply]
+  convert Complex.reCLM.summable this
+
+lemma Complex.summable_nat_multipliable_one_add (f : ℕ → ℂ) (hf : Summable f)
+    (hff : ∀ n : ℕ, 1 + f n ≠ 0) : Multipliable (fun n : ℕ => 1 + f n) := by
+  refine Complex.multipliable_of_summable_log  (fun n => 1 + f n) (by simpa) ?_
+  simpa only [forall_const] using Complex.log_of_summable hf
```

Comment:
If `hff` fails then surely the result is trivially true: it has infinite product 0. Or am I missing something basic?

Reply:
Yes I think it should also be true, but I don't think we have results about vanishing/non-vanishing of infinite products yet. I do plan to add these later, so I've added this to the TODO list.